lib/ResultAggregator.ts

```diff
@@ -123,3 +141,9 @@ export class ResultAggregator implements IResultAggregator {
 export interface IResultAggregator {
   aggregateResults: (results: IResult[]) => IAggregateResult[];
 }
+
+export interface IProcessedTimestamps {
+  maxTs: number[];
```

Comment:
Could we use `Time` instead of `T` in variable name, for consistency with the rest of the code's varnames?

Reply:
Yes I could, but to be clear, maxTs refers to the maximum timestamp, not time. I agree that the capital T looks weird here, but I wanted to keep with the javascript style.

Would you prefer maxTimestamps or something similar?

Reply:
What about `timestampsAverage`, `timestampsMax` and `timestampsMin`?

Reply:
> Yes I could, but to be clear, maxTs refers to the maximum timestamp, not time. I agree that the capital T looks weird here, but I wanted to keep with the javascript style.

Aha, well, that illustrates my point on why we should expand the `T` :-)

Reply:
Indeed, changed it to Johnie's suggestions